pkg/controller/machine/controller.go

```diff
@@ -421,61 +466,77 @@ func isInvalidMachineConfigurationError(err error) bool {
 	return false
 }
 
-func (r *ReconcileMachine) setPhase(machine *machinev1.Machine, phase string, failureCause error) error {
+func (r *ReconcileMachine) updateStatus(machine *machinev1.Machine, phase string, failureCause error, originalConditions []machinev1.Condition) error {
```

Comment:
WDYT about call `machine.GetConditions` inside this function and don't pass it here explicitly? As I see we return value from `GetConditions` method.

Reply:
The reason we pass it explicitly here is because we need to catch what the original state of the conditions is before we make any modifications to it. So getting the conditions once they've been passed into this function wouldn't work as we have already modified them on the machine by the time they get to this point.

I admit this isn't the easiest way this could be factored, but I think to make it simpler we would need to undertake quite a large refactor of the reconciliation logic to ensure we only update the status in one place - maybe a follow up for the not too distant future 🤔

Reply:
maybe drop a todo comment in the code to keep track?